core/src/main/java/org/elasticsearch/action/support/nodes/BaseNodesResponse.java

```diff
@@ -91,11 +105,36 @@ public TNodeResponse getAt(int position) {
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         clusterName = ClusterName.readClusterName(in);
+        failures = in.readArray(FailedNodeException[]::new, FailedNodeException::new);
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         clusterName.writeTo(out);
+        out.writeArray(failures);
     }
+
+    /**
+     * Expected to be invoked by any {@code BaseNodesResponse} that implements {@code ToXContent} at the top level of the response object.
+     */
+    protected XContentBuilder toInnerXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
I really like the concept of the `"_nodes"`, but I'm not so sure about the name. Practically every `ToXContent` impl that I noticed had a `"nodes"` block. I'm a bit worried by how close those are.

How does that make you feel about `_nodes`? The problem is that I don't have a better name because that's the right one. :)